src/library-authoring/component-info/ComponentUsage.tsx

```diff
@@ -0,0 +1,102 @@
+
+import AlertError from '../../generic/alert-error';
+import Loading from '../../generic/Loading';
+import { useFetchIndexDocuments } from '../../search-manager';
+import { useComponentDownstreamLinks } from '../data/apiHooks';
+import messages from './messages';
+
+interface ComponentUsageProps {
+  usageKey: string;
+}
+
+type ComponentUsageTree = Record<string, {
+  key: string,
+  contextName: string,
+  links: {
+    usageKey: string,
+    displayName: string,
+    url: string,
+  }[]
+}>;
+
+const getContainerUrl = (contextKey: string, usageKey: string) => (
+  `${getPath(getConfig().PUBLIC_PATH)}course/${contextKey}/container/${usageKey}`
+);
+
+export const ComponentUsage = ({ usageKey }: ComponentUsageProps) => {
+  const {
+    data: dataDownstreamLinks,
+    isError: isErrorDownstreamLinks,
+    error: errorDownstreamLinks,
+    isLoading: isLoadingDownstreamLinks,
+  } = useComponentDownstreamLinks(usageKey);
+
+  const downstreamKeys = dataDownstreamLinks || [];
+
+  const {
+    data: downstreamHits,
+    isError: isErrorIndexDocuments,
+    error: errorIndexDocuments,
+    isLoading: isLoadingIndexDocuments,
+  } = useFetchIndexDocuments({
+    filter: [`usage_key IN ["${downstreamKeys.join('","')}"]`],
+    limit: downstreamKeys.length,
+    attributesToRetrieve: ['usage_key', 'breadcrumbs', 'context_key'],
+    enabled: !!downstreamKeys.length,
+  });
+
+  if (isErrorDownstreamLinks || isErrorIndexDocuments) {
+    return <AlertError error={errorDownstreamLinks || errorIndexDocuments} />;
+  }
+
+  if (isLoadingDownstreamLinks || (isLoadingIndexDocuments && !!downstreamKeys.length)) {
+    return <Loading />;
+  }
+
+  if (!downstreamKeys.length || !downstreamHits) {
+    return <FormattedMessage {...messages.detailsTabUsageEmpty} />;
+  }
+
+  const componentUsage = downstreamHits.reduce<ComponentUsageTree>((acc, hit) => {
+    const link = hit.breadcrumbs.at(-1);
+    // istanbul ignore if: this should never happen. it is a type guard for the breadcrumb last item
+    if (!(link && ('usageKey' in link))) {
+      return acc;
+    }
+
+    const linkData = {
+      ...link,
+      url: getContainerUrl(hit.contextKey, link.usageKey),
+    };
+
+    if (hit.contextKey in acc) {
+      acc[hit.contextKey].links.push(linkData);
+    } else {
+      acc[hit.contextKey] = {
+        key: hit.contextKey,
+        contextName: hit.breadcrumbs[0].displayName,
+        links: [linkData],
+      };
+    }
+    return acc;
+  }, {});
+
+  const componentUsageList = Object.values(componentUsage);
+
+  return (
+    <>
+      {
+        componentUsageList.map((context) => (
+          <Collapsible key={context.key} title={context.contextName} styling="basic">
+            {context.links.map(({ usageKey: downstreamUsageKey, displayName, url }) => (
+              <Hyperlink key={downstreamUsageKey} destination={url} target="_blank">{displayName}</Hyperlink>
+            ))}
+          </Collapsible>
```

Comment:
We need to adjust the CSS to arrange these links properly.

![image](https://github.com/user-attachments/assets/43423ab8-b716-4539-8ead-269048700f9b)


Ignore duplicate course names, I just have multiple courses with same name. :sweat_smile:

Reply:
Fixed!